From ba09742a3f30947b5336b167ac703247506ac3a7 Mon Sep 17 00:00:00 2001
From: Tzafrir Cohen <tzafrir.cohen@xorcom.com>
Date: Wed, 22 Mar 2017 15:45:32 +0200
Subject: [PATCH] xpp: prevent dracut from loading Astribank drivers

Prevent an early load of the Astribank (xpp) drivers at initrd time, as
they require some files from the root filesystem.

This installs a file in the dracut configuration directory that will
take effect the next time dracut generates an initrd image.

This installs an configuration file for dracut which is ignored if
dracut is not installed and is a no-op if xpp_usb is not loaded on your
system.

Signed-off-by: Tzafrir Cohen <tzafrir.cohen@xorcom.com>
---
 xpp/50-dahdi.conf | 5 +++++
 xpp/Makefile.am   | 3 +++
 2 files changed, 8 insertions(+)
 create mode 100644 xpp/50-dahdi.conf

diff --git a/xpp/50-dahdi.conf b/xpp/50-dahdi.conf
new file mode 100644
index 0000000..a0a5ce8
--- /dev/null
+++ b/xpp/50-dahdi.conf
@@ -0,0 +1,5 @@
+# A dracut.conf(5) snippet
+# Disable loading the Astribank (xpp_usb) drivers. This is because
+# it relies on files in the root filesystem:
+# See https://issues.asterisk.org/jira/browse/DAHLIN-352
+omit_drivers+=" xpp_usb "
diff --git a/xpp/Makefile.am b/xpp/Makefile.am
index 05f1b1c..56ce64e 100644
--- a/xpp/Makefile.am
+++ b/xpp/Makefile.am
@@ -62,6 +62,9 @@ dist_sbin_SCRIPTS	= $(perl_scripts)
 man_MANS		+= $(perl_mans)
 CLEANFILES		= $(perl_checks) $(perl_mans)
 
+dracutconfdir = @prefix@/lib/dracut/dracut.conf.d
+dracutconf_DATA = 50-dahdi.conf
+
 if	PBX_USB
 
 SUBDIRS		+= oct612x xtalk
-- 
2.11.0

